CMakeLists.txt

```diff
@@ -25,6 +25,7 @@ SET(CMAKE_Fortran_ABI_COMPILED 0)
 SET(CMAKE_Fortran_COMPILER_SUPPORTS_F90 1)
 SET(CMAKE_Fortran_PREPROCESS_SOURCE
     "<CMAKE_Fortran_COMPILER> -cpp <DEFINES> <INCLUDES> <FLAGS> -E <SOURCE> -o <PREPROCESSED_SOURCE>")
+set(CMAKE_Fortran_MODDIR_FLAG "-module ")
```

Comment:
@xoviat, https://gitlab.kitware.com/cmake/cmake/issues/17597 was because this was missing and the compiler identification is partial when flang is not fully built. When flang is fully built, this flag is set by CMake.

Reply:
I also received the "old or corrupt module file" messages that you received here, caused by CMake not deleting the module file when it rebuilds the associated object. Any idea how to solve that?